### PR TITLE
feat: support `required` in belongs_to relationship

### PR DESCRIPTION
**Pull request by transhapHigsn**
### Contributor checklist

- [x] My commit messages follow the [Conventional Commit Message Format](https://gist.github.com/stephenparish/9941e89d80e2bc58a153#format-of-the-commit-message)
      For example: `fix: Multiply by appropriate coefficient`, or
      `feat(Calculator): Correctly preserve history`
      Any explanation or long form information in your commit message should be
      in a separate paragraph, separated by a blank line from the primary message
- [ ] Bug fixes include regression tests
- [x] Features include unit/acceptance tests

### CODE REVIEW

**Comment by transhapHigsn** [comment]
Test cases not completed yet.

**Comment by transhapHigsn** [comment]
Changes for issue #38

File: lib/ash_json_api/json_schema/json_schema.ex [hunk 1]
```diff
@@ -683,6 +684,17 @@ defmodule AshJsonApi.JsonSchema do
     end)
   end
 
+  defp required_relationship_attributes(resource, accept) do
+    resource
+    |> AshJsonApi.Resource.fields()
+    |> Enum.filter(&(is_nil(accept) || &1 in accept))
+    |> Enum.filter(fn field ->
+      rel = Ash.Resource.relationship(resource, field)
+      !is_nil(rel) && rel.required?
```
Comment (zachdaniel) [inline lib/ash_json_api/json_schema/json_schema.ex:693]: Some relationships don't have have the `required?` field, so we'll want to use `!is_nil(rel) &&  Map.get(rel, :required?)`

**Comment by zachdaniel** [comment]
The changes look great, added a comment on the implementation. Two things: 1.) we'll want to remove the `mix.lock` changes, and 2.) need to add some tests.

Then it is good to go! Great job 👍

File: lib/ash_json_api/json_schema/json_schema.ex (hunk 1 above)
Comment (transhapHigsn) [reply lib/ash_json_api/json_schema/json_schema.ex:693]: Ok.

**Comment by transhapHigsn** [comment]
@zachdaniel yeah will do that.

File: test/acceptance/post_test.exs [hunk 2]
```diff
@@ -73,8 +73,8 @@ defmodule Test.Acceptance.PostTest do
     end
 
     attributes do
-      attribute(:id, :uuid, primary_key?: true)
-      attribute(:name, :string)
+      attribute(:id, :uuid, primary_key?: true, allow_nil?: false)
```
Comment (zachdaniel) [inline test/acceptance/post_test.exs:76]: do we need the changes in this file?

File: test/acceptance/belongs_to_required_test.exs [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        Api
+        |> post("/posts", %{
+          data: %{
+            type: "post",
+            attributes: %{
+              id: id,
+              name: "Invalid Post 1"
+            },
+            relationships: %{
+              author: %{
```
Comment (zachdaniel) [inline test/acceptance/belongs_to_required_test.exs:126]: Try this with
```
author: nil
```

This request should actually have been rejected as invalid.

File: test/acceptance/post_test.exs (hunk 2 above)
Comment (transhapHigsn) [reply test/acceptance/post_test.exs:76]: not exactly, wanted to keep them as reference. will remove if you want that.

File: test/acceptance/belongs_to_required_test.exs (hunk 3 above)
Comment (transhapHigsn) [reply test/acceptance/belongs_to_required_test.exs:126]: Have tried that, for current code and `nil` , response is same.
```
resp_body: %{
    "errors" => [
      %{
        "code" => "FrameworkError",
        "detail" => "Returned when an unexpected error in the framework has occured.",
        "id" => "c8fe0c27-6cff-46c1-8761-d06d3c1fc042",
        "title" => "Framework Error"
      }
    ],
    "jsonapi" => %{"version" => "1.0"}
  },
```

File: test/acceptance/post_test.exs (hunk 2 above)
Comment (zachdaniel) [reply test/acceptance/post_test.exs:76]: Its fine to leave it 👍

File: test/acceptance/belongs_to_required_test.exs (hunk 3 above)
Comment (zachdaniel) [reply test/acceptance/belongs_to_required_test.exs:126]: I'll have to solve that problem before we merge this then, but aside from that everything looks good.

File: test/acceptance/post_test.exs (hunk 2 above)
Comment (transhapHigsn) [reply test/acceptance/post_test.exs:76]: ok

File: test/acceptance/belongs_to_required_test.exs (hunk 3 above)
Comment (transhapHigsn) [reply test/acceptance/belongs_to_required_test.exs:126]: ok

Comment (transhapHigsn) [reply test/acceptance/belongs_to_required_test.exs:126]: Let me know if i can do anything to help.

**Pull request merged**